android/tangram/src/com/mapzen/tangram/MapController.java

```diff
@@ -184,6 +184,9 @@ void dispose() {
         queueEvent(new Runnable() {
             @Override
             public void run() {
+                for (Map.Entry<String, MapData> entry : clientDataSources.entrySet()) {
```

Comment:
Can you also do a `checkPointer(mapPointer)` when map is init. Line: 179.

P.S. Github did not allow me to comment of that specific line!!